src/ASM/LR/ASMu3D.C

```diff
@@ -2498,6 +2501,7 @@ void ASMu3D::generateThreadGroupsFromElms (const IntVec& elms)
   for (int elm : elms)
     if (this->getElmIndex(elm+1) > 0)
       myElms.push_back(this->getElmIndex(elm+1)-1);
+  hasElms = true;
```

Comment:
An alternative (more elegant?, at least with less footprint) solution could be to add, e.g. 0 or -1 as a value in myElms here, such that it is never empty if we do generateThreadGroupsFromElms and the test in ASMu3D::integrate will still work.

Reply:
The you don't need the hasElms member

Reply:
Okay, did this instead.